This fixes various problems with fr30-elf and likely msp430-elf too
diff --git a/gcc/config/elfos.h b/gcc/config/elfos.h
index acb376283cf..09dd4c89985 100644
--- a/gcc/config/elfos.h
+++ b/gcc/config/elfos.h
@@ -214,8 +214,8 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
    The definitions say how to change sections to the .init and .fini
    sections.  This is the same for all known svr4 assemblers.  */
 
-#define INIT_SECTION_ASM_OP	"\t.section\t.init"
-#define FINI_SECTION_ASM_OP	"\t.section\t.fini"
+#define INIT_SECTION_ASM_OP	"\t.section\t.init,\"ax\""
+#define FINI_SECTION_ASM_OP	"\t.section\t.fini,\"ax\""
 
 /* Output assembly directive to move to the beginning of current section.  */
 #ifdef HAVE_GAS_SUBSECTION_ORDERING

diff --git a/gcc/testsuite/lib/prune.exp b/gcc/testsuite/lib/prune.exp
index 04c6a1dd7a1..64c9f56fbf0 100644
--- a/gcc/testsuite/lib/prune.exp
+++ b/gcc/testsuite/lib/prune.exp
@@ -85,7 +85,8 @@ proc prune_gcc_output { text } {
     # Ideally the tests would indicate that executable stacks were needed
     # to the linker.  But the option for that varies and may not even exist
     # on some targets.  So we're stuck pruning the warning.
-    regsub -all "(^|\n)(\[^\n\]*: warning:\[^\n\]*requires executable stack\[^\n\]*\n?)+" $text "\\1" text
+    regsub -all "(^|\n)(\[^\n\]*: warning:\[^\n\]*executable stack\[^\n\]*\n?)+" $text "\\1" text
+    regsub -all "(^|\n)(\[^\n\]*: warning:\[^\n\]*has a LOAD segment\[^\n\]*\n?)+" $text "\\1" text
 
     # Ignore harmless warnings from Xcode 3.2.x.
     regsub -all "(^|\n)\[^\n\]*ld: warning: can't add line info to anonymous symbol\[^\n\]*" $text "" text
